halo2_proofs/src/poly/domain.rs

```diff
@@ -2,24 +2,31 @@
 //! domain that is of a suitable size for the application.
 
 use crate::{
-    arithmetic::{best_fft, parallelize},
-    plonk::Assigned,
+    arithmetic::{self, best_fft, parallelize},
+    fft::{
+        recursive::{self, FFTData},
+    },
+    multicore,
+    plonk::{get_duration, get_time, start_measure, stop_measure, Assigned, log_info},
 };
 
 use super::{Coeff, ExtendedLagrangeCoeff, LagrangeCoeff, Polynomial, Rotation};
 use ff::WithSmallOrderMulGroup;
 use group::{
-    ff::{BatchInvert, Field, PrimeField},
+    ff::{self, BatchInvert, Field, PrimeField},
     Group,
 };
 
-use std::marker::PhantomData;
+use std::{env::var, marker::PhantomData};
+
+/// TEMP
+pub static mut FFT_TOTAL_TIME: usize = 0;
 
 /// This structure contains precomputed constants and other details needed for
 /// performing operations on an evaluation domain of size $2^k$ and an extended
 /// domain of size $2^{k} * j$ with $j \neq 0$.
 #[derive(Clone, Debug)]
-pub struct EvaluationDomain<F: Field> {
+pub struct EvaluationDomain<F: ff::Field> {
```

Comment:
This change necessary?

Reply:
Not anymore,  good catch!  I had two conflicting `Field` types and prefixed them to stay sane.